### PR TITLE
Profiles: handle set primary name different flows

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-####

## What changed (plus any additional context for devs)

- condition to know if the ens is the primary name wasn't working correctly, `isPrimaryName` is refactored to get the ens resolved address and compare it to current wallet to know if it's the wallet's primary name, before it was checking if it was a primary name for a wallet, which could be other one
- new flow from @ibrahimtaveras00 , setting primary name wasn't updating the wallet avatar, now we'll get the new ens name and the new avatar at the same time after "setting primary name"

https://user-images.githubusercontent.com/12115171/169401959-ec54862a-a945-4845-b5a0-6a604377b297.mp4

- refactored handlers/ens to use`getProviderForNetwork` instead of `web3Provider`

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/redux/wallets.js [hunk 1]
```diff
@@ -234,10 +232,7 @@ export const fetchWalletNames = () => async (dispatch, getState) => {
       const visibleAccounts = filter(wallet.addresses, 'visible');
       return map(visibleAccounts, async account => {
         try {
-          const ens = await lookupAddressWithRetry(
```
Comment (estebanmino) [inline src/redux/wallets.js:237]: updated this method to use the `fetchReverseRecord` coming from profiles

**Review by benisgold — APPROVED** [review]

File: src/hooks/useENSProfile.ts [hunk 2]
```diff
@@ -47,5 +55,7 @@ export default function useENSProfile(
     [accountAddress, data?.owner?.address]
   );
 
-  return { data, isLoading, isOwner, isSuccess };
+  const isPrimaryName = isLowerCaseMatch(walletNames[accountAddress], name);
```
Comment (jxom) [inline src/hooks/useENSProfile.ts:58]: prefer native `.toLowerCase()`? 

`isLowerCaseMatch` uses lodash, and i think we are trying to migrate away from that.

Comment (estebanmino) [reply src/hooks/useENSProfile.ts:58]: oh

File: src/hooks/useENSProfile.ts [hunk 3]
```diff
@@ -42,10 +48,24 @@ export default function useENSProfile(
...
-    () => isLowerCaseMatch(data?.owner?.address || '', accountAddress),
-    [accountAddress, data?.owner?.address]
-  );
+  const isOwner =
+    data?.owner?.address?.toLocaleLowerCase() === accountAddress?.toLowerCase();
+
+  // if a ENS NFT is sent, the ETH coinAddress record doesn't change
+  // if the user tries to use it to set primary name the tx will go through
+  // but the name won't be set. Disabling it to avoid these cases
+  const isSetNameEnabled =
```
Comment (estebanmino) [inline src/hooks/useENSProfile.ts:57]: important fix

File: src/hooks/useWalletENSAvatar.ts [hunk 4]
```diff
@@ -14,46 +12,11 @@ export default function useWalletENSAvatar() {
...
-    let updatedWallets;
-    for (const key of walletKeys) {
-      const wallet = wallets[key];
-      for (const account of wallet?.addresses) {
-        const ens = walletNames[account.address];
-        if (ens) {
-          const images = await fetchImages(ens);
-          if (images?.avatarUrl) {
-            let avatarChanged = false;
-            const addresses = wallet.addresses.map((acc: RainbowAccount) => {
```
Comment (estebanmino) [inline src/hooks/useWalletENSAvatar.ts:27]: this was iterating though all `wallet.addresses` for each `account of wallet?.addresses` producing that the wallet avatars weren't being set correctly, seen by @ibrahimtaveras00 

image

File: src/redux/wallets.js [hunk 5]
```diff
@@ -224,6 +227,54 @@ export const createAccountForWallet = (id, color, name) => async (
...
+            images?.avatarUrl !== account?.image
+              ? images?.avatarUrl
+              : account.image,
+        });
+      } else {
+        addresses.push(account);
+      }
+    }
+    // don't update wallets if nothing changed
+    if (avatarChanged) {
```
Comment (estebanmino) [inline src/redux/wallets.js:259]: https://github.com/rainbow-me/rainbow/pull/3374/files#r877656269

now we go though the addresses once and set the avatarUrl to the correct account

**Pull request merged**